app/controllers/solidus_stripe/webhooks_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require "solidus_stripe/webhook/event"
+require "stripe"
+
+module SolidusStripe
+  class WebhooksController < Spree::BaseController
+    SIGNATURE_HEADER = "HTTP_STRIPE_SIGNATURE"
+
+    skip_before_action :verify_authenticity_token, only: :create
+
+    respond_to :json
+
+    def create
+      request.body.read
```

Comment:
Thoughts on moving this to a service? I read your point that we are not ready with the infrastructure, and that makes sense. But when we will, and want to convert this, it will probably be easier because we can deprecate that class. 

No strong preference here: in the @elia's PR there are several controllers not using services, so this is more a general discussion about the approach we want to take.

Reply:
Some of this could probably be kept inside `SolidusStripe::Webhook::Event`, something like:

```rb
def (SolidusStripe::Webhook::Event).from_request(request)
  tolerance = SolidusStripe.configuration.webhook_signature_tolerance
  secret = Rails.application.credentials.solidus_stripe[:webhook_endpoint_secret] # move to a configuration?
  signature_header = request.headers[SIGNATURE_HEADER]

  Stripe::Webhook.construct_event(payload, signature_header, secret, tolerance: tolerance)
end
```

Reply:
Moving things to services requires some decisions to be made, like the interface we create for a result object. I think it doesn't pay off to build a skeleton here that will need to be deprecated later. At most, as @elia suggested, we can try to move some things into collaborators. The `from_request` method makes sense, but I'd also move the creation to the `SolidusStripe::Webhook::Event` there. This way, an instance of the collaborator will be involved. I'd mark the method as `@api private` so we're free to remove it later. Thoughts?

Reply:
I think I need to see the code in order to understand fully the proposal, but I'm down with it from what I understand… and it's way easier to go from `@api private` to `@api public` than the other way around 🙌 

The main goal is to have building block that are easy to reuse in case the target app wants (or needs) to customize or redo the controller.

Reply:
Please, take a look now. I moved the logic other than publishing the event to a class method in `SolidusStripe::Webhook::Event`.